File: leetcode/53/b.py
from typing import List

class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
      thissum = 0
      maxsum = float("-inf")
      for i in range(0, len(nums)):
        thissum = 0
        for j in range(i, len(nums)):
          thissum += nums[j]
          if thissum > maxsum:
            maxsum = thissum
      return maxsum


solution = Solution()
print(solution.maxSubArray([-2,1,-3,4,-1,2,1,-5,4]))

File: leetcode/53/b.js
function maxSubArray(nums) {
  let thissum, i, j, maxsum = 0;
  for (i = 0; i < nums.length; i++) {
    thissum = 0;
    for (j = i; j < nums.length; j++) {
      thissum += nums[j];
      if (thissum > maxsum) {
        maxsum = thissum;
      }
    }
  }
  return maxsum;
}

console.log(maxSubArray([-2,1,-3,4,-1,2,1,-5,4]));